node14-typescript.js

```diff
@@ -0,0 +1,99 @@
+'use strict';
+
+module.exports = {
+  extends: require.resolve('./node14.js'),
+  overrides: [
+    {
+      files: ['**/*.ts'],
+      extends: [
+        "plugin:@typescript-eslint/eslint-recommended",
+        'plugin:@typescript-eslint/recommended'
+      ],
+      parser: '@typescript-eslint/parser',
+      parserOptions: {
+        ecmaVersion: 2020,
+        sourceType: 'module',
+        /*
+        NOTE: You will need to define your eslint config as .eslint.js and add snippet below to the config:
+        'parserOptions': {
+          'tsconfigRootDir': __dirname
+        }
+        */
+        project: '.tsconfig'
+      },
+      plugins: ['@typescript-eslint'],
+      rules: {
+        'prefer-arrow-callback': [
+          'error',
+          {
+            allowUnboundThis: true
+          }
+        ],
+        curly: 'error',
+        'node/no-unsupported-features/es-syntax': 'off',
+        'no-array-constructor': 'off', // must be disabled so the typescript one can take over.
+        'no-dupe-keys': 'error',
+        'no-extra-semi': 'off', // must be disabled so the typescript one can take over.
+        'no-restricted-imports': [
+          'error',
+          {
+            patterns: ['*/config']
+          }
+        ],
```

Comment:
We might want to omit this. It's likely a project-specific configuration.

Reply:
Good point, removed